src/python/pants/backend/codegen/protobuf/python/rules.py

```diff
@@ -95,13 +97,18 @@ async def generate_python_from_protobuf(
         target_stripped_sources_request,
     )
 
+    lockfile_hex_digest = None
+    if python_protobuf_mypy_plugin.lockfile != "<none>":
+        lockfile_request = await Get(PythonLockfileRequest, MypyProtobufLockfileSentinel())
+        lockfile_hex_digest = lockfile_request.hex_digest
```

Comment:
This code is very redundant...

Could `PexRequest` take the sentinel type + lockfile and make this request for you instead? Or could the sentinel type be converted into a wrapper around the filename (i.e., no longer just a sentinel)?

Or, more fundamentally, should `PexRequest.requirements` become a union of `PexRequirements | PythonToolRequirementsBase`?

Reply:
I think that we will absolutely want to fix this, but it's fine in a followup.

Reply:
Yeah I recommend we wait to fix this. For example, https://github.com/pantsbuild/pants/issues/12542 will make lots of changes to these call sites.

Reply:
Yeah, I don't particularly like this code, and I'd be happy to take suggestions on how to make the improvement.